plugin/src/main/java/io/fabric8/maven/plugin/enricher/SelectorVisitor.java

```diff
@@ -92,6 +91,44 @@ public void visit(DeploymentSpecBuilder item) {
         }
     }
 
+    static public class PetSet extends SelectorVisitor<PetSetSpecBuilder> {
+
+        public PetSet(EnricherManager enricherManager) {
+            super(enricherManager);
+        }
+
+        @Override
+        public void visit(PetSetSpecBuilder item) {
+            Map<String, String> selectorMatchLabels =
+                    KubernetesResourceUtil.removeVersionSelector(enricherManager.extractSelector(getConfig(), Kind.REPLICATION_CONTROLLER));
+            final io.fabric8.kubernetes.api.model.LabelSelector selector = item.getSelector();
+            if (selector == null) {
+                item.withNewSelector().addToMatchLabels(selectorMatchLabels).endSelector();
+            } else {
+                selector.getMatchLabels().putAll(selectorMatchLabels);
+            }
+        }
+    }
+
+    static public class DaemonSet extends SelectorVisitor<DaemonSetSpecBuilder> {
+
+        public DaemonSet(EnricherManager enricherManager) {
+            super(enricherManager);
+        }
+
+        @Override
+        public void visit(DaemonSetSpecBuilder item) {
+            Map<String, String> selectorMatchLabels =
+                    KubernetesResourceUtil.removeVersionSelector(enricherManager.extractSelector(getConfig(), Kind.REPLICATION_CONTROLLER));
```

Comment:
shoud be `Kind.DAEMON_SET` (btw, there is an error for the `Deployment` selector, I will fix that.